src/mono/mono/utils/lifo-semaphore.c

```diff
@@ -0,0 +1,89 @@
+#include <mono/utils/lifo-semaphore.h>
+
+LifoSemaphore *
+mono_lifo_semaphore_init (void)
+{
+	LifoSemaphore *semaphore = g_new0 (LifoSemaphore, 1);
+	if (semaphore == NULL)
```

Comment:
I think you have to use `try_new` or somesuch, which might not exist, to actually get null, vs. abort.
Systematic problem in runtime imho -- doesn't want to handle out of memory in many places.